blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md

```diff
@@ -0,0 +1,797 @@
+        .newFreelancer(_image, _name, _title, _description, _price)
+        .send({ from: address });
+      getFreelancers();
+    } catch (error) {
+      alert(error);
+    }
+  };
+
+  const changeDescription = async (_index, _description) => {
+    try {
+      await contract.methods
+        .changeDescription(_index, _description)
+        .send({ from: address });
+      getFreelancers();
+    } catch (error) {
+      alert(error);
+    }
+  };
+
+  const changePrice = async (_index, _newprice) => {
+    try {
+      await contract.methods
+        .changePrice(_index, _newprice)
+        .send({ from: address });
+      getFreelancers();
+    } catch (error) {
+      alert(error);
+    }
+  };
+
+  const toggleAvailable = async (_index) => {
+    try {
+      await contract.methods
+        .toggleAvailable(_index)
+        .send({ from: address });
+      getFreelancers();
+    } catch (error) {
+      alert(error);
+    }
+  };
+
+  const hireFreelancer = async (_index, _hours) => {
+    try {
+      const cUSDContract = new kit.web3.eth.Contract(IERC, cUSDContractAddress);
+      const totalPayment = _hours * freelancers[_index].price;
+      const cost = new BigNumber(totalPayment)
+              .shiftedBy(ERC20_DECIMALS)
+              .toString();
+      console.log(totalPayment);
+      await cUSDContract.methods
+        .approve(contractAddress, cost)
+        .send({ from: address });
+      await contract.methods
+        .hireFreelancerHourly(_index, _hours, cost)
+        .send({ from: address });
+      getFreelancers();
+      getBalance();
+      alert("you have successfully sent cusd to this user");
+    } catch (error) {
+      alert(error);
+    }
+  };
+
+  useEffect(() => {
+    connectToWallet();
+  }, []);
+
+  useEffect(() => {
+    if (kit && address) {
+      getBalance();
+    }
+  }, [kit, address, getBalance]);
+
+  useEffect(() => {
+    if (contract) {
+      getFreelancers();
+    }
+  }, [contract, getFreelancers]);
+
+  return (
+    <div className="App">
+      <Home cUSDBalance={cUSDBalance} addFreelancer={addFreelancer} />
+      <Freelancers
+        freelancers={freelancers}
+        hireFreelancer={hireFreelancer}
+        changeDescription={changeDescription}
+        changePrice={changePrice}
+        walletAddress={address}
+        toggleAvailable={toggleAvailable}
+      />
+    </div>
+  );
+}
+
+export default App;
+```
+
+### Breakdown
+
+The first few lines of the code import the required dependencies and components.
+
+```javascript
+import "./App.css";
+import Home from "./components/home";
+import { Freelancers } from "./components/Freelancers";
+import { useState, useEffect, useCallback } from "react";
+import Web3 from "web3";
+import { newKitFromWeb3 } from "@celo/contractkit";
+import freelancer from "./contracts/freelancer.abi.json";
+import IERC from "./contracts/IERC.abi.json";
+import BigNumber from "bignumber.js";
+
+```
+
+#### Declaring constants and initializing state:
```

Comment:
can you make this a double hashtag or triple at most or maybe just use a bold text